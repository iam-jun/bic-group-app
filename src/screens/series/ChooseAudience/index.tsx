import React, { useEffect, useMemo } from 'react';
import {
  View, StyleSheet, Keyboard,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import { isEmpty, isEqual } from 'lodash';

import Header from '~/beinComponents/Header';
import SelectAudience, { ContentType } from '~/components/SelectAudience';
import useSelectAudienceStore from '~/components/SelectAudience/store';
import { useBaseHook } from '~/hooks';
import { useBackPressListener, useRootNavigation } from '~/hooks/navigation';
import { CreationSeriesProps } from '~/interfaces/ISeries';
import seriesStack from '~/router/navigator/MainStack/stacks/series/stack';
import { getAudienceIdsFromAudienceObject } from '~/screens/articles/CreateArticle/helper';
import useSeriesStore, { ISeriesState } from '../store';
import useModalStore from '~/store/modal';

const ChooseSeriesAudience = ({ route }: CreationSeriesProps) => {
  const { isFirstStep, isEditAudience, initAudienceGroups = [] } = route?.params || {};

  const { t } = useBaseHook();
  const theme: ExtendedTheme = useTheme();
  const styles = createStyle(theme);
  const { showAlert } = useModalStore((state) => state.actions);

  const { rootNavigation } = useRootNavigation();

  const seriesActions = useSeriesStore((state: ISeriesState) => state.actions);
  const { loading } = useSeriesStore((state: ISeriesState) => state);

  const selectingAudienceIds = useSelectAudienceStore((state) => state.selectedIds);
  const selectingAudienceGroups = useSelectAudienceStore((state) => state.selectedAudiences.groups);
  const selectAudienceActions = useSelectAudienceStore((state) => state.actions);
  const selectAudienceResetStore = useSelectAudienceStore((state) => state.reset);

  const setInitDataToSelectingAudiences = () => {
    const newSelectingGroups = {};
    initAudienceGroups?.forEach((group) => {
      newSelectingGroups[group?.id] = group;
    });
    selectAudienceActions.setSelectedAudiences(newSelectingGroups);
  };

  useEffect(() => {
    if (initAudienceGroups?.length > 0) {
      setInitDataToSelectingAudiences();
    }
  }, [initAudienceGroups]);

  useEffect(() => {
    if (isFirstStep) selectAudienceResetStore();
  }, []);

  const initAudienceIds = useMemo(
    () => getAudienceIdsFromAudienceObject({ groups: initAudienceGroups }), [initAudienceGroups],
  );

  const isAudienceUpdated = !isEqual(initAudienceIds, selectingAudienceIds);
  const isAudienceSelected = !(isEmpty(selectingAudienceIds?.groupIds) && isEmpty(selectingAudienceIds?.userIds));

  const disabled = !(isAudienceUpdated && isAudienceSelected) || loading;

  const handleBack = () => {
    if (!disabled || isAudienceUpdated) {
      Keyboard.dismiss();
      showAlert({
        title: t('discard_alert:title'),
        content: t('discard_alert:content'),
        cancelBtn: true,
        cancelLabel: t('common:btn_discard'),
        confirmLabel: t('common:btn_stay_here'),
        onCancel: () => {
          setInitDataToSelectingAudiences();
          rootNavigation.goBack();
        },
      });
      return;
    }
    rootNavigation.goBack();
  };

  useBackPressListener(handleBack);

  const handleSave = () => {
    if (isFirstStep) {
      seriesActions.setAudience(selectingAudienceIds);
      seriesActions.setAudienceGroups(selectingAudienceGroups);
      rootNavigation.replace(
        seriesStack.createSeries,
      );
    } else if (isEditAudience) {
      if (isAudienceUpdated) {
        showAlert({
          title: t('post:create_post:title_audience_changed'),
          content: t('post:create_post:text_discard_change_audience'),
          cancelBtn: true,
          cancelLabel: t('common:btn_discard'),
          confirmLabel: t('post:create_post:btn_save_change'),
          onConfirm: () => {
            seriesActions.setAudience(selectingAudienceIds);
            seriesActions.setAudienceGroups(selectingAudienceGroups);
            rootNavigation.goBack();
          },
        });
      } else {
        seriesActions.setAudience(selectingAudienceIds);
        seriesActions.setAudienceGroups(selectingAudienceGroups);
        rootNavigation.goBack();
      }
    } else {
      seriesActions.setAudience(selectingAudienceIds);
      seriesActions.setAudienceGroups(selectingAudienceGroups);
      rootNavigation.goBack();
    }
  };

  return (
    <View style={styles.container}>
      <Header
        title={t('post:title_post_to')}
        buttonProps={{
          disabled,
          loading,
          testID: 'series_select_audience.btn_next',
        }}
        buttonText={t('common:btn_next')}
        onPressButton={handleSave}
        onPressBack={handleBack}
      />
      <SelectAudience contentType={ContentType.SERIES} />
    </View>
  );
};

const createStyle = (theme: ExtendedTheme) => {
  const { colors } = theme;
  return StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: colors.neutral,
    },
  });
};

export default ChooseSeriesAudience;
