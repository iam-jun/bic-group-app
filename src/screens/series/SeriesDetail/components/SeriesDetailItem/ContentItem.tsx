/* eslint-disable @typescript-eslint/no-unused-vars */
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import React, { FC } from 'react';
import { View, StyleSheet } from 'react-native';
import { Avatar, Button } from '~/baseComponents';
import { useRootNavigation } from '~/hooks/navigation';
import { IPost, PostType } from '~/interfaces/IPost';
import mainTabStack from '~/router/navigator/MainStack/stack';
import { spacing } from '~/theme';
import Text from '~/baseComponents/Text';
import Image from '~/beinComponents/Image';
import { borderRadius } from '~/theme/spacing';
import ViewSpacing from '~/beinComponents/ViewSpacing';
import images from '~/resources/images';
import DeactivatedView from '~/components/DeactivatedView';
import { escapeMarkDown } from '~/utils/formatter';
import { getSummaryPostItemInSeires, getTitlePostItemInSeries } from '~/helpers/common';

type ContentItemProps = {
  item: IPost;
};

const ContentItem: FC<ContentItemProps> = ({
  item,
}) => {
  const {
    actor, coverMedia, summary, content, media,
  } = item || {};
  const { avatar, fullname, isDeactivated } = actor || {};
  const { url } = coverMedia || {};
  const { images: imagesPost } = media || {};
  const coverUrlItem = item?.type === PostType.ARTICLE ? url : imagesPost?.[0]?.url;
  const escapeMarkDownContent = escapeMarkDown(content);
  const titlePost = getTitlePostItemInSeries(escapeMarkDownContent);
  const summaryPost = getSummaryPostItemInSeires(escapeMarkDownContent, titlePost);

  const summaryItem = item?.type === PostType.ARTICLE ? summary : summaryPost;

  const theme = useTheme();
  const { colors } = theme;
  const styles = createStyle(theme);
  const { rootNavigation } = useRootNavigation();

  const onPressActor = () => {
    if (!actor?.id || isDeactivated) return;

    const payload = { userId: actor.id };
    rootNavigation.navigate(mainTabStack.userProfile, payload);
  };

  const colorFullName = isDeactivated ? colors.grey40 : colors.neutral60;

  return (
    <View>
      <View style={styles.row}>
        <View style={styles.actor}>
          <Button
            style={styles.avatarContainer}
            onPress={onPressActor}
          >
            <Avatar.Small style={styles.avatar} isRounded source={avatar} />
            <ViewSpacing width={spacing.margin.small} />
            <View style={styles.fullnameContainer}>
              <Text.BodyXSMedium color={colorFullName} numberOfLines={1}>
                {fullname}
              </Text.BodyXSMedium>
              {isDeactivated && <DeactivatedView style={styles.deactivatedView} />}
            </View>
          </Button>
        </View>
        {!!coverUrlItem ? <Image style={styles.img} source={coverUrlItem} />
          : <Image style={styles.img} source={images.no_image_avalable} />}
      </View>
      {!!summaryItem && (
        <View style={styles.summaryView}>
          <Text.ParagraphM color={colors.neutral40} numberOfLines={4}>
            { summaryItem }
          </Text.ParagraphM>
        </View>
      )}
    </View>
  );
};

const createStyle = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    row: {
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'flex-start',
    },
    actor: {
      flex: 1,
      flexDirection: 'row',
      marginRight: spacing.margin.small,
    },
    img: {
      width: 120,
      height: 80,
      borderRadius: borderRadius.small,
    },
    summaryView: {
      marginTop: spacing.margin.large,
    },
    avatarContainer: {
      flex: 1,
      flexDirection: 'row',
      alignItems: 'center',
    },
    avatar: {
      alignSelf: 'flex-start',
    },
    fullnameContainer: {
      flex: 1,
    },
    deactivatedView: {
      marginTop: spacing.margin.tiny,
    },
  });
};

export default ContentItem;
