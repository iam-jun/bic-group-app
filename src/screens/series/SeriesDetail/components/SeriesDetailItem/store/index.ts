import IBaseState from '~/store/interfaces/IBaseState';
import { createStore } from '~/store/utils';
import deleteItemFromSeriesDetail from './actions/deleteItemFromSeriesDetail';

export interface ISeriesDetailItemState extends IBaseState{
  actions?: {
    deleteItemFromSeriesDetail: (seriesId: string, itemId: string) => void;
  }
}

const useSeriesDetailItem = (set, get) => ({
  actions: {
    deleteItemFromSeriesDetail: deleteItemFromSeriesDetail(set, get),
  },
});

const useSeriesDetailItemStore = createStore<ISeriesDetailItemState>(useSeriesDetailItem);

export default useSeriesDetailItemStore;
