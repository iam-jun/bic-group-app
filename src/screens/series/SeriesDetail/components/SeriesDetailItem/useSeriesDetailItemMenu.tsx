import React from 'react';
import i18next from 'i18next';
import seriesStack from '~/router/navigator/MainStack/stacks/series/stack';

import { useRootNavigation } from '~/hooks/navigation';
import { Button } from '~/baseComponents';
import useSeriesDetailItemStore, { ISeriesDetailItemState } from './store';
import useModalStore from '~/store/modal';
import { IPost, PostType } from '~/interfaces/IPost';
import SeriesContentModal from '~/components/series/SeriesContentModal';
import getAudienceListWithNoPermission from '~/store/permissions/actions/getAudienceListWithNoPermission';
import { PermissionKey } from '~/constants/permissionScheme';
import homeStack from '~/router/navigator/MainStack/stacks/homeStack/stack';

const useSeriesDetailItemMenu = (
  seriesId: string, item: IPost,
) => {
  const { id, type, audience } = item;
  const { rootNavigation } = useRootNavigation();
  const {
    showAlert, hideBottomList, showBottomList, showModal,
  } = useModalStore((state) => state.actions);

  const actions = useSeriesDetailItemStore((state: ISeriesDetailItemState) => state.actions);
  const typeRemove = type === PostType.ARTICLE ? 'article' : 'post';

  const groupAudience = audience?.groups || [];
  const audienceListCannotPinContent = getAudienceListWithNoPermission(
    groupAudience,
    [
      PermissionKey.FULL_PERMISSION,
      PermissionKey.PIN_CONTENT,
    ],
  );

  const onPress = () => {
    hideBottomList();
    showAlert({
      title: i18next.t(`post:series_detail_item_menu:delete_${typeRemove}`),
      content: i18next.t(`series:text_remove_${typeRemove}`),
      cancelBtn: true,
      confirmLabel: i18next.t('common:text_remove'),
      ConfirmBtnComponent: Button.Danger,
      onConfirm: () => {
        actions.deleteItemFromSeriesDetail(seriesId, id);
      },
      confirmBtnProps: { type: 'ghost' },
    });
  };

  const onPressReorder = () => {
    hideBottomList();
    rootNavigation.navigate(seriesStack.reorderItems, { seriesId });
  };

  const onPressViewSeries = () => {
    hideBottomList();

    showModal({
      isOpen: true,
      isFullScreen: true,
      titleFullScreen: i18next.t('common:btn_view_series'),
      ContentComponent: <SeriesContentModal id={id} />,
    });
  };

  const onPressPin = () => {
    hideBottomList();
    rootNavigation?.navigate?.(homeStack.pinContent, { postId: id });
  };

  const defaultData = [
    {
      id: 1,
      testID: 'series_detail_item_menu.reoder',
      title: i18next.t('post:series_detail_item_menu:reorder'),
      onPress: onPressReorder,
    },
    {
      id: 2,
      testID: 'series_detail_item_menu.delete',
      title: i18next.t(`post:series_detail_item_menu:delete_${typeRemove}`),
      onPress,
    },
    {
      id: 3,
      testID: 'series_detail_item_menu.view_series',
      title: i18next.t('post:series_detail_item_menu:view_series'),
      onPress: onPressViewSeries,
    },
    {
      id: 4,
      testID: 'series_detail_item_menu.pin',
      title: i18next.t('common:pin_unpin'),
      shouldBeHidden:
        audienceListCannotPinContent.length === groupAudience.length,
      onPress: onPressPin,
    },
  ];

  const showMenu = () => {
    showBottomList({ isOpen: true, data: defaultData } as any);
  };

  return {
    showMenu,
  };
};

export default useSeriesDetailItemMenu;
