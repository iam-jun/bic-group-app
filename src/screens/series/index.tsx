export { default as SeriesItem } from './components/SeriesItem';
export { default as SeriesDetail } from './SeriesDetail';
export { default as ChooseSeriesAudience } from './ChooseAudience';
export { default as CreateStories } from './CreateSeries';
