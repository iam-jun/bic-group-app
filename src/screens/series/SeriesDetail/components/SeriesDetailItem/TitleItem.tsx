import { ExtendedTheme, useTheme } from '@react-navigation/native';
import React, { FC } from 'react';
import { View, StyleSheet, Text as RNText } from 'react-native';
import Text from '~/baseComponents/Text';
import { IPost, PostType } from '~/interfaces/IPost';
import { spacing, dimension } from '~/theme';
import { formatNumberWithZeroPrefix, escapeMarkDown } from '~/utils/formatter';
import { Button } from '~/baseComponents';
import useSeriesDetailItemMenu from './useSeriesDetailItemMenu';
import { useBaseHook } from '~/hooks';

type TitleItemProps = {
    index: number;
    item: IPost;
    seriesId: string;
    isActor: boolean;
}

const TitleItem: FC<TitleItemProps> = ({
  index, item, seriesId, isActor,
}) => {
  const { title, id, content } = item || {};
  const { t } = useBaseHook();
  const theme = useTheme();
  const { colors } = theme;
  const styles = createStyle(theme);
  const titleItem = item?.type === PostType.ARTICLE ? title : escapeMarkDown(content);

  const { showMenu } = useSeriesDetailItemMenu(seriesId, id, item?.type);

  return (
    <View style={styles.container}>
      <View style={styles.titleView}>
        <Text.H1 color={colors.neutral20}>{formatNumberWithZeroPrefix(index)}</Text.H1>
        <View style={styles.slash} />
        <View style={{ flex: 1 }}>
          {!!titleItem ? <Text.H3 numberOfLines={1} color={colors.neutral80}>{ titleItem }</Text.H3>
            : (
              <RNText
                numberOfLines={1}
                style={styles.noContent}
              >
                { t('series:text_no_content') }
              </RNText>
            )}
        </View>
      </View>
      {isActor
      && (
      <Button.Raise
        icon="menu"
        size="small"
        testID="content_header.menu"
        onPress={showMenu}
      />
      )}
    </View>
  );
};

const createStyle = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'space-between',
    },
    slash: {
      marginHorizontal: spacing.margin.small,
      borderLeftWidth: 1,
      borderLeftColor: colors.neutral20,
      height: 44,
      transform: [
        {
          rotate: '15deg',
        },
      ],
    },
    titleView: {
      flex: 1,
      marginRight: spacing.margin.small,
      flexDirection: 'row',
      alignItems: 'center',
    },
    noContent: {
      fontStyle: 'italic',
      fontWeight: '600',
      fontSize: dimension?.sizes.h3,
      color: colors.neutral30,
    },
  });
};

export default TitleItem;
