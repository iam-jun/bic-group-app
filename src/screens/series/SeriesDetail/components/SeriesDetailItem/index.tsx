import { ExtendedTheme, useTheme } from '@react-navigation/native';
import React, { FC } from 'react';
import { StyleSheet } from 'react-native';
import { IPost, PostType } from '~/interfaces/IPost';
import { spacing } from '~/theme';
import TitleItem from './TitleItem';
import ViewSpacing from '~/beinComponents/ViewSpacing';
import ContentItem from './ContentItem';
import { useRootNavigation } from '~/hooks/navigation';
import articleStack from '~/router/navigator/MainStack/stacks/articleStack/stack';
import homeStack from '~/router/navigator/MainStack/stacks/homeStack/stack';
import { Button } from '~/baseComponents';
import { TrackingEventContentReadAction, TrackingEventContentReadProperties, TrackingEventType } from '~/interfaces/ITrackingEvent';
import { trackEvent } from '~/services/tracking';

type SeriesDetailItemProps = {
    item: IPost;
    index: number;
    seriesId: string;
    isActor: boolean;
}

const SeriesDetailItem: FC<SeriesDetailItemProps> = ({
  item, index, seriesId, isActor,
}) => {
  const theme = useTheme();
  const styles = createStyle(theme);

  const { rootNavigation } = useRootNavigation();

  const goToArticleContentDetail = () => {
    rootNavigation.navigate(articleStack.articleContentDetail, { articleId: item?.id });

    // tracking event
    const eventContentReadProperties: TrackingEventContentReadProperties = {
      content_type: PostType.ARTICLE,
      action: TrackingEventContentReadAction.SERIES_ITEM,
    };
    trackEvent({ event: TrackingEventType.CONTENT_READ, properties: eventContentReadProperties });
  };

  const goToPostDetail = () => {
    rootNavigation.navigate(homeStack.postDetail, { post_id: item?.id });

    // tracking event
    const eventContentReadProperties: TrackingEventContentReadProperties = {
      content_type: PostType.POST,
      action: TrackingEventContentReadAction.SERIES_ITEM,
    };
    trackEvent({ event: TrackingEventType.CONTENT_READ, properties: eventContentReadProperties });
  };

  const onRedirect = () => {
    if (item?.type === PostType.ARTICLE) {
      goToArticleContentDetail();
    } else {
      goToPostDetail();
    }
  };

  return (
    <Button
      style={styles.container}
      onPress={onRedirect}
      testID="series_detail.list_item.item"
    >
      <TitleItem index={index} item={item} seriesId={seriesId} isActor={isActor} />
      <ViewSpacing height={spacing.margin.extraLarge} />
      <ContentItem item={item} />
    </Button>
  );
};

const createStyle = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      padding: spacing.padding.large,
      backgroundColor: colors.white,
      borderBottomWidth: 1,
      borderBottomColor: colors.gray1,
    },
  });
};

export default SeriesDetailItem;
