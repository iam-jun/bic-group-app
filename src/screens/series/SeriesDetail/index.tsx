import { ExtendedTheme, useTheme } from '@react-navigation/native';
import React, { useCallback, useEffect } from 'react';
import { Keyboard, StyleSheet, View } from 'react-native';
import { useDispatch } from 'react-redux';
import { Button } from '~/baseComponents';
import Header from '~/beinComponents/Header';
import { BottomListProps } from '~/components/BottomList';
import { useBaseHook } from '~/hooks';
import { useUserIdAuth } from '~/hooks/auth';
import { useRootNavigation } from '~/hooks/navigation';
import { useMyPermissions } from '~/hooks/permissions';
import { IAudienceGroup } from '~/interfaces/IPost';
import AlertDeleteAudiencesConfirmContent from '~/components/posts/AlertDeleteAudiences';
import usePostsStore from '~/store/entities/posts';
import postsSelector from '~/store/entities/posts/selectors';
import modalActions from '~/storeRedux/modal/actions';
import DeletedItem from '../../../components/series/DeletedItem';
import { getSeriesMenu } from '../../../helpers/series';
import SeriesDetailHeader from './components/SeriesDetailHeader';
import useSeriesStore, { ISeriesState } from '../store';

const SeriesDetail = ({ route }: any) => {
  const { params } = route || {};
  const { seriesId } = params || {};
  const theme = useTheme();
  const styles = createStyle(theme);
  const userId = useUserIdAuth();
  const dispatch = useDispatch();
  const { t } = useBaseHook();
  const { rootNavigation } = useRootNavigation();

  const series = usePostsStore(useCallback(postsSelector.getPost(seriesId, {}), [seriesId]));

  const {
    actor, id, deleted, audience,
  } = series;
  const actions = useSeriesStore((state: ISeriesState) => state.actions);

  useEffect(() => {
    actions.getSeriesDetail(seriesId);
  }, []);

  const { hasPermissionsOnAtLeastOneScope, PERMISSION_KEY }
    = useMyPermissions();
  const canDeleteOwnPost = hasPermissionsOnAtLeastOneScope(
    'groups',
    audience?.groups,
    PERMISSION_KEY.GROUP.DELETE_OWN_POST,
  );

  const handleError = (listIdAudiences: string[]) => {
    if (listIdAudiences?.length <= 0 || audience?.groups?.length <= 0) {
      return;
    }

    const listAudiences = listIdAudiences.map((audienceId) => {
      const _audience = audience.groups.find(
        (audience: IAudienceGroup) => audience?.id === audienceId,
      );
      return _audience;
    });
    if (canDeleteOwnPost) {
      dispatch(
        modalActions.showAlert({
          title: t('series:title_delete_audiences_of_series'),
          children: (
            <AlertDeleteAudiencesConfirmContent
              data={listAudiences}
              textContent={t('series:content_delete_audiences_of_series')}
            />
          ),
          cancelBtn: true,
          confirmLabel: t('common:text_remove'),
          ConfirmBtnComponent: Button.Danger,
          onConfirm: () => actions.removeAudiences(id, listIdAudiences),
          confirmBtnProps: { type: 'ghost' },
        }),
      );
    } else {
      dispatch(
        modalActions.showAlert({
          title: t('series:title_delete_audiences_of_series'),
          children: (
            <AlertDeleteAudiencesConfirmContent
              data={listAudiences}
              textContent={t('series:content_not_able_delete_of_series')}
            />
          ),
          cancelBtn: true,
          cancelLabel: t('common:btn_close'),
          onConfirm: null,
        }),
      );
    }
  };

  const handleConfirmDelete = () => {
    actions.deleteSeries(id, handleError);
  };

  const onRightPress = () => {
    Keyboard.dismiss();
    const data = getSeriesMenu({
      reactionsCount: {},
      isActor: actor?.id == userId,
      dispatch,
      seriesId: id,
      navigaton: rootNavigation,
      isFromDetail: true,
      handleConfirmDelete,
    });

    dispatch(
      modalActions.showBottomList({ isOpen: true, data } as BottomListProps),
    );
  };

  if (deleted) {
    return (
      <View style={styles.container}>
        <Header />
        <DeletedItem />
      </View>
    );
  }

  return (
    <View style={styles.container}>
      <Header
        rightIcon="menu"
        onRightPress={onRightPress}
      />

      <SeriesDetailHeader series={series} />
      {/* for the next sprint */}
      {/* list SeriesDetailArticleItem */}
      {/* <SeriesDetailArticleItem
        index={1}
        article={article_from_series}
      /> */}
    </View>
  );
};

const createStyle = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: colors.gray5,
    },
  });
};

export default SeriesDetail;
