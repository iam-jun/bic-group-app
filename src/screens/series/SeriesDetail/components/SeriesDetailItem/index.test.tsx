import * as React from 'react';
import { article } from '~/test/mock_data/series';
import { renderWithRedux } from '~/test/testUtils';
import SeriesDetailItem from '.';

describe('SeriesDetailItem component', () => {
  it('render correctly', () => {
    const wrapper = renderWithRedux(<SeriesDetailItem
      index={1}
      item={article}
      seriesId="5264f1b3-c8b8-428a-9fb8-7f075f03d0c8"
      isActor
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
