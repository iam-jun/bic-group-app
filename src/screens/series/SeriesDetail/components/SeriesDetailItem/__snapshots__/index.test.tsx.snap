// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeriesDetailItem component render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#F2F5F8",
      "borderBottomWidth": 1,
      "opacity": 1,
      "padding": 16,
    }
  }
  testID="series_detail.list_item.item"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "marginRight": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#181C32",
              "fontFamily": "BeVietnamPro-SemiBold",
              "fontSize": 28,
            },
            Object {
              "color": "#BAC0DE",
            },
            undefined,
          ]
        }
      >
        01
      </Text>
      <View
        style={
          Object {
            "borderLeftColor": "#BAC0DE",
            "borderLeftWidth": 1,
            "height": 44,
            "marginHorizontal": 8,
            "transform": Array [
              Object {
                "rotate": "15deg",
              },
            ],
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "#9FA6D1",
              "fontSize": 20,
              "fontStyle": "italic",
              "fontWeight": "600",
            }
          }
        >
          No content
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "borderRadius": 6,
          },
          undefined,
        ]
      }
      testID="content_header.menu"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 6,
              "elevation": 2,
              "flexDirection": "row",
              "justifyContent": "center",
              "shadowColor": "rgba(0, 0, 0, 1)",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.12,
              "shadowRadius": 1.5,
            },
            undefined,
            Object {
              "backgroundColor": "#FFFFFF",
              "height": 28,
              "paddingHorizontal": 8,
            },
          ]
        }
        testID="button.content"
      >
        <View
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          hitSlop={
            Object {
              "bottom": 10,
              "left": 10,
              "right": 10,
              "top": 10,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                undefined,
                Object {
                  "backgroundColor": undefined,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              <View
                testID="button.icon"
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "height": 24,
        "paddingHorizontal": undefined,
        "paddingVertical": undefined,
        "width": undefined,
      }
    }
  />
  <View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "marginRight": 8,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "opacity": 1,
            }
          }
          testID="button_wrapper"
        >
          <View
            style={
              Array [
                Object {
                  "borderRadius": 6,
                  "height": 32,
                  "width": 32,
                },
                Object {
                  "alignSelf": "flex-start",
                },
              ]
            }
            testID="avatar"
          >
            <View
              style={
                Array [
                  Array [
                    Object {
                      "borderRadius": 6,
                      "height": 32,
                      "width": 32,
                    },
                    Object {
                      "borderRadius": 16,
                    },
                  ],
                  Object {},
                ]
              }
              testID="avatar_container"
            >
              <View
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Array [
                      Array [
                        Object {
                          "borderRadius": 6,
                          "height": 32,
                          "width": 32,
                        },
                        Object {
                          "borderRadius": 16,
                        },
                      ],
                      Object {
                        "borderColor": "#F2F5F8",
                        "borderWidth": 2,
                      },
                      Object {
                        "backgroundColor": "#FFFFFF",
                      },
                    ],
                  ]
                }
              >
                <FastImageView
                  onFastImageError={[Function]}
                  resizeMode="cover"
                  source={
                    Object {
                      "uri": "https://bein-entity-attribute-stg.s3.ap-southeast-1.amazonaws.com/user/avatar/images/original/e69febe3-5962-41f9-9258-3a8579f3b0ab",
                    }
                  }
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                  testID="avatar.image"
                />
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "height": undefined,
                "paddingHorizontal": undefined,
                "paddingVertical": undefined,
                "width": 8,
              }
            }
          />
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#212745",
                    "fontFamily": "BeVietnamPro-Medium",
                    "fontSize": 11,
                  },
                  Object {
                    "color": "#282E53",
                  },
                  undefined,
                ]
              }
            >
              Wall-E
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
              "height": 80,
              "width": 120,
            },
          ]
        }
      >
        <FastImageView
          onFastImageError={[Function]}
          resizeMode="cover"
          source={
            Object {
              "testUri": "../../../assets/images/no_image_avalable.png",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
